nodes/modifier_change/extrude_separate.py

```diff
@@ -195,16 +195,13 @@ def process(self):
 
         meshes = match_long_repeat([vertices_s, edges_s, faces_s, masks_s, heights_s, scales_s, matrixes_s, face_data_s])
 
-        for vertices, edges, faces, masks, heights, scales, matrixes, face_data in zip(*meshes):
+        for vertices, edges, faces, masks_, heights_, scales_, matrixes_, face_data in zip(*meshes):
 
             new_extruded_faces = []
             new_extruded_faces_append = new_extruded_faces.append
-            fullList(heights, len(faces))
-            fullList(scales, len(faces))
-            fullList(matrixes, len(faces))
-            fullList(masks, len(faces))
+            heights, scales, matrixes, masks = make_repeaters([heights_, scales_, matrixes_, masks_])
```

Comment:
Shouldn't this depend on len(faces)?

Reply:
It creates iterators that later are used in zip(faces, repeater), so depends of faces but does not create the full list
```
def make_repeaters(lists):
    chain = itertools.chain
    repeat = itertools.repeat
    out =[]
    for l in lists:
        out.append(chain(l, repeat(l[-1])))

    return out
```